### DISCUSSION TITLE
Can not edit asset account as it complains about IBAN already in use

### DISCUSSION THREAD

**Discussion by jurgenhaas** [firefly-iii/firefly-iii#7326] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

I've created an asset account and didn't bother about the "Opening balance" to start with. I then imported all the bookings which went great. Not I want to edit the account and input the opening balance, but I can't save the form as I get the error `It looks like this IBAN is already in use.`

### Debug information

Debug information generated at 2023-04-03 00:00:00 Europe/Berlin for Firefly III version **6.0.6**.

| Scope | Version |
| --- | --- |
| Firefly III | 6.0.6 |
| Firefly III API | 2.0.1 |
| Build | 441, 01-04-2023 08:04:21 CEST |
| DB version | 19 (exp. 19) |
| PHP | `8.2.4` |
| Host | `Linux` |

| System info | Value |
| --- | --- |
| System TZ | Europe/Berlin |
| Browser TZ | Europe/Berlin |
| App environment | local |
| App debug mode | false |
| App cache driver | file |
| App logging | notice, stack |
| Display errors | Off |
| Error reporting | ALL errors |
| Max upload | 67108864 (64 MB) |
| Interface | apache2handler |
| Default language | de_DE |
| Default locale | equal |
| BCscale | 12 |
| DB drivers | mysql, pgsql, sqlite |
| Current driver | mysql |
| Trusted proxies (.env) | `**` |

| User info | Value |
| --- | --- |
| Login provider | eloquent  |
| User guard | web  |
| Headers | N/A, N/A  |
| Stateful domains |  |
| Session start | 2023-04-01 00:00:00 |
| Session end | 2023-04-30 23:59:59 |
| Session first | 2022-03-24 00:00:00 |
| User ID | 1 |
| User language | en_GB |
| User locale | en_GB |
| Attempt at "en_GB.utf8" | 'en_GB.utf8' |
| Attempt at "en_GB.UTF-8" | 'en_GB.UTF-8' |
| User agent | Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/111.0 |

**Comment by JC5** [comment]
This may happen when you have another asset account with that same IBAN. In the past, sometimes bugs would allow duplicate IBANs. But this is no longer the case.

Remove the IBAN from this asset account or from the other asset account(s) with this IBAN.

**Reply by jurgenhaas** [reply]
Ah, while importing bookings it created an income account with the same IBAN. When I delete the IBAN there, the problem disappeared. Is that really how things should work, i.e. creating a new income account which is the same as an existing asset account?
